avocado/plugins/resolvers.py

```diff
@@ -98,6 +119,16 @@ def resolve(reference):
                                PythonUnittestResolver._find_compat)
 
 
+class PythonUnittestDiscoverer(Discoverer):
```

Comment:
I understand the reasons that you have created a new class. But having Resolver, Discoverer and Runner for each type sounds too much IMO. Specially when two of them have only one method each (resolve and discover).

Are you planing having other methods to those *Discoverer* classes? A possible solution would be aggregate most of the methods in one single class. But I understand that this will change too much the current design. Wondering what do you think about this.

And, no worries here, I'm just thinking out of loud.

Reply:
Yes I thought about that, But then I decedent for a new class, because IMHO it doesn't make a sense to use Resolver without a reference. I mean, what the Resolver should be resolving? I think we have here two different use-cases, first for resolving references and second for test search, and because of  preservation of high cohesion I decided to create a new class. Also, the Discoverer is optional, you don't have to use it if you don't want to search for tests without reference. 

But we can discuss that, because it isn't problem to change that.

Reply:
My opinion here is that the somewhat "implicit" resolution of tests done with the `datadir.paths.test_dir` is legacy is should remain like that -- better not add it to the resolver code.

This would lead to implementation of `Discoverer` **only** when there's actual no reference, somewhat achieving what @beraldoleal commented here.